nixos/modules/services/networking/thelounge.nix

```diff
@@ -88,7 +138,25 @@ in
     systemd.services.thelounge = {
       description = "The Lounge web IRC client";
       wantedBy = [ "multi-user.target" ];
-      preStart = "ln -sf ${pkgs.writeText "config.js" configJsData} ${dataDir}/config.js";
+      preStart = ''
+        set -o errexit -o pipefail -o nounset -o errtrace
+        shopt -s inherit_errexit
+
+        ln -sf ${pkgs.writeText "config.js" configJsData} ${dataDir}/config.js
+
+        mkdir ${dataDir}/users
+
+        ${concatStrings (mapAttrsToList (user: options:
+          let
+            path = "${dataDir}/users/${user}.json";
+          in
+          ''
+            if [ ! -e ${path} ]; then
+              ${pkgs.jq}/bin/jq --null-input --arg password ${if options.hashedPassword != null then escapeShellArg options.hashedPassword else "\"$(<${escapeShellArg options.hashedPasswordFile})\""} \
```

Comment:
Passing the password as a command line argument will leak it through the process' `/proc/<pid>/cmdline` file. This doesn't really matter for `hashedPassword`, since that will be available through the Nix store anyway, but I'm guessing the point of `hashedPasswordFile` is additional security, in which case it does matter. You could instead provide the hashed passwords to `jq` through environment variables and access them with `$ENV.<env_var_name>` in your filter.

Reply:
Good catch, thanks!